hnn_core/network.py

```diff
@@ -1253,23 +1253,50 @@ def add_connection(self, src_gids, target_gids, loc, receptor,
 
         self.connectivity.append(deepcopy(conn))
 
-    def clear_connectivity(self):
-        """Remove all connections defined in Network.connectivity
+    def clear_connectivity(self, src_types=None):
+        """Remove connections with src_type in Network.connectivity.
+
+        Parameters
+        ----------
+        src_types : list | None
+            Source types of connections to be removed
+            None - Remove all connections other than those which have an
+            external drive as a source type
```

Comment:
Yes @jasmainak I was trying to preserve the old logic. If no argument is passed to `clear_connectivity` then the default value of src_types should be called "All" instead of "None".  Will refactor the code accordingly and make the required changes.